### PR TITLE
Add support for KMS

### PR DESCRIPTION
**Pull request by felddy**
Supports setting per-region KMS aliases.  The helper script now pre-processes the `packer.json` file instead of modifying the environment variables used by packer.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
I think this is an excellent solution to this problem - bravo! 🔑 🌐

**Review by jsf9k — APPROVED** [review]
Strong work, good sir!

**Comment by felddy** [comment]
I've made a few substantial changes to support @jsf9k 's standing request to be able to build images outside of Travis.  I realized I neglected this use case when I updated the `README.md`.  

I'm going to re-request reviews due to the size of these changes.

**Review by jsf9k — APPROVED** [review]
Thank you for taking manual builds into consideration.

File: README.md [hunk 1]
```diff
@@ -38,38 +38,61 @@ Here is an example of encrypting the credentials for Travis:
...
+1. **Non-release test**: After a normal commit or pull request Travis
+   will build the project, and run tests and validation on the
+   packer configuration.  It will __not__ build an image.
+1. **Pre-release deploy**: Publish a GitHub release
+   with the "This is a pre-release" checkbox checked.  An image will be built
+   and deployed to the single region defined by the `PACKER_BUILD_REGION`
+   environment variable.
+1. **Production release deploy**: Publish a GitHub release with
+   the "This is a pre-release" checkbox unchecked.  An image will be built
+   in the `PACKER_BUILD_REGION` and copied to each regions listed in the
```
Comment (dav3r) [inline README.md]: You had it right the first time- it should be `region`, not `regions` here.

File: README.md [hunk 2]
```diff
@@ -38,38 +38,61 @@ Here is an example of encrypting the credentials for Travis:
...
-  - `PACKER_PRE_RELEASE`: a boolean tag applied to the final image
 
-Here is an example of how to kick off a build:
+Here is an example of how to kick off a pre-release build:
 
 ```console
+pip install --requirement requirements-dev.txt
 export PACKER_BUILD_REGION="us-east-2"
-export PACKER_DEPLOY_REGIONS="us-east-1,us-west-1,us-west-2"
+export PACKER_DEPLOY_REGION_KMS_MAP="us-east-1:alias/cool/ebs,us-east-2:alias/cool/ebs,us-west-1:alias/cool/ebs,us-west-2:alias/cool/ebs"
```
Comment (dav3r) [inline README.md:83]: If I was doing a manual pre-release build, I'd just do this:
```
export PACKER_BUILD_REGION="us-east-2"
export PACKER_DEPLOY_REGION_KMS_MAP="us-east-2:alias/cool/ebs"
```
There's no need to put more than one region in `PACKER_DEPLOY_REGION_KMS_MAP` if the other regions will just be ignored.  In the Travis config file, I get that it's a different story, but here in the README, this is what folks who are doing a manual build will crib from.

**Review by dav3r — APPROVED** [review]
Approved, but please take a look at my comments. 👍

File: README.md (hunk 2 above)
Comment (felddy) [reply README.md:83]: Well that example serves a few purposes.  It shows the format of the map variable outside of `.travis` and by changing the `pre-release` command to `release` you get a full deploy.   Let me see if I can clear that up.

File: README.md (hunk 1 above)
Comment (felddy) [reply README.md]: Been looking at this too long.  Good catch:
"unfixed" in 454dc76

File: README.md (hunk 2 above)
Comment (felddy) [reply README.md:83]: Clarified in 454dc76

Comment (dav3r) [reply README.md:83]: Works for me 👍

**Review by dav3r — APPROVED** [review]
Still approved! 🚀

**Pull request merged**